pkg/model/pki.go

```diff
@@ -120,6 +120,16 @@ func (b *PKIModelBuilder) Build(c *fi.ModelBuilderContext) error {
 		c.AddTask(t)
 	}
 
+	if b.Cluster.Spec.KubeAPIServer.ProxyClientCertFile != nil && b.Cluster.Spec.KubeAPIServer.ProxyClientKeyFile != nil {
```

Comment:
I actually think we should create this based on the kubernetes version i.e. for versions >= 1.8 (or maybe >= 1.7), we create the keys.  Or we just always create the key, but only set it on >= 1.7.  There is a `VersionGTE` helper which you can `git grep` for similar usage.